### PR TITLE
Big List of Fixes

### LINKED ISSUE #3211: In disgrace all are equal - +1 characteristic
**Describe the bug**

Hi all, thank you for the blackshields addition which is fab. First error I noticed is for the "in disgrace all are equal", which has that +1 to one of several characteristics. In new recruit it forces you to pick one t apply to all characters:

Screenshot_20240528-195403

However, according to rule each character should be able to pick it individually what that +1 is:

Screenshot_20240528-195407

I hope this is possible to fix - thank you!

### LINKED ISSUE #3212: Kharybdis Assault Claw Should not be hidden in The Covenant Of Fire (SAL)
When creating a Salamanders army list using the Covenant of Fire Right of War list the Kharybdis Assault Claw is hidden when it shouldn’t be.

This Right of War restricts units that _must_ deep strike, and it’s optional for the Kharybdis.

### LINKED ISSUE #3213: Points calculation error on Rapier Carriers entry
**Describe the bug**
The points for:

1. Upgrading a single Rapier Carrier
2. Adding additional shell options

Is currently being tripled e.g. an upgrade to a Quad launcher is 60 points rather than 20 points

**To Reproduce**
1. Create a HH list for Legiones Astartes
4. Add a Rapier Carrier to the list
5. Select an upgrade option for the carrier 
6. The points for the upgrade are triple what they upgrade should cost

**Expected behavior**
I'd expect the upgrade costs to match the values stated in the published army list

**Device and data (please complete the following information):**
 - OS: iOS and 
 - New Recruit
 - New Recruit v258.6
 - Data release version Iron Warriors V40

**Additional context**
None

### LINKED ISSUE #3216: Jarl hidden for praetor
> Hello! It looks like the latest update to Horus Heresy has caused the Space Wolf Praetor to lose the 'Jarl' upgrade. 
> It was there as of a few nights ago, but when I checked today, it was listed as an error, "Jarl is hidden" in the build list.

image

Looks like we're unhiding it if there are no instances of blackshields in the force, and space wolves in parent (instead of roster).
image

### LINKED ISSUE #3220: Cult Arcana Discipline hidden for terminator praetor
Cult arcana discipline is checking "XV thousand sons in parent" and not in force.
Regular praetor has a second copy of that condition, which might also work around the blackshields logic. 
image

### LINKED ISSUE #3229: Primarch entry not showing for Custodes
Primarch entry missing from the Custodes Crusade force, making it impossible to select Constantin Valdor

v19.0.3

### LINKED ISSUE #3230: unable to give Centurions a Psychic Discipline in Thousand Sons Forces
As the title suggests I seem unable to give Centurions a Psychic Discipline in Thousand Sons Forces on new recuit, it allows minor arcana but not the option to add the psychic powers to them across the armour types 

I admit I am rather new to it so maybe a me thing

### LINKED ISSUE #3234: Duplicate Optae in AL HQ data
**Describe the bug**
18 Identical unit entries for Optae appearing in the HQ category for the Alpha Legion data
image

**To Reproduce**
1. Go to LA -XX: Alpha Legion new Crusade FOC
2. See multiple Optae in HQ section on left

**Expected behavior**
Only one Optae unit entry in HQs

**Screenshots**

**Device and data (please complete the following information):**
 - OS: PC Windows 10
 - App being used to create lists: Battlescribe
 - App version: 2.03.21
 - Data release version v19.0.3

**Additional context**

### LINKED ISSUE #3235: Horus Aximand both in AL data and allowed for Loyalist SoH
**Describe the bug**
Horus Aximand both in AL data and allowed for Loyalist SoH
image
image

**To Reproduce**
1. Go to LA -XX: Alpha Legion new Crusade FOC
2. See Horus Aximand in HQ section

1. Go to LA -XVI: Sons of Horus new Crusade FOC
2. Set Allegiance to Loyalist
3. See Horus Aximand in HQ section

**Expected behavior**
Horus Aximand should only be available to Traitor Sons of Horus

**Screenshots**

**Device and data (please complete the following information):**
 - OS: PC Windows 10
 - App being used to create lists: Battlescribe
 - App version: 2.03.21
 - Data release version v19.0.3

**Additional context**

### PR DESCRIPTION
**Pull request by The4D6**
Please set a title for the PR describing your changes,
fill out the information below, and delete this heading.

## Fixed Units
- Rapier Battery costs now calculate per-battery not per-model (as the latter counts the gunners)

### Related Issues
* Closes #3235 
* Closes #3234
* Closes #3230 
* Closes #3229 
* Closes #3220 
* Closes #3216
* Closes #3213 
* Closes #3212
* Closes #3211 

## Not done yet

- [x] 3235
- [x] 3234
- [x] 3230
- [x] 3229
- [x] 3220
- [x] 3216
- [x] 3213
- [x] 3212
- [x] 3211
- [ ] 3210
- [ ] 3187 - NSH this doesn't look done (but I could be wrong)
- [ ] 3186
- [ ] 3180
- [ ] 3162

### CODE REVIEW

**Comment by The4D6** [comment]
PR is big enough as is, would like to get this in before making more changes

File: 2022 - Horus Heresy.gst [hunk 1]
```diff
@@ -247,7 +247,7 @@ During Reactions made in any Phase, a unit equipped with Jump PAcks may not acti
         </rule>
         <rule name="Infantry Unit Type" hidden="false" id="ed36-77cb-5da7-3298" publicationId="e77a-823a-da94-16b9" page="195">
           <alias>
-            <undefined>Infantry</undefined>
```
[the remark this answers is not in the document]
Comment (nstephenh) [reply 2022 - Horus Heresy.gst:250]: Added a future fix. Though for some reason the editor keeps adding the undefined tag.

File: 2022 - Horus Heresy.gst [hunk 2]
```diff
@@ -9125,6 +9150,7 @@ Hull Mounted (Rear) Mounted Twin-linked Heavy Bolter or Heavy Flamers</character
           <categoryLinks>
             <categoryLink targetId="9f1e-fbf0-3032-32fe" id="8af4-11d7-46c2-b6c4" name="Building Sub-type" primary="false"/>
             <categoryLink targetId="a24f-12d8-36c1-f477" id="d87b-579e-4808-a9de" name="Fortification:" primary="false"/>
```
Comment (nstephenh) [inline 2022 - Horus Heresy.gst]: Fortification the force org slot should be removed (I've done it).

File: 2022 - Horus Heresy.gst [hunk 3]
```diff
@@ -265,18 +265,30 @@ During Reactions made in any Phase, a unit equipped with Jump PAcks may not acti
...
           <modifiers>
             <modifier type="set" field="hidden" value="true">
-              <conditions>
-                <condition field="selections" scope="force" value="0" percentValue="false" shared="true" includeChildSelections="true" includeChildForces="false" childId="a2cf-c464-bfdd-4467" type="equalTo"/>
-              </conditions>
+              <conditionGroups>
+                <conditionGroup type="or">
+                  <conditions>
+                    <condition type="equalTo" value="0" field="selections" scope="force" childId="a2cf-c464-bfdd-4467" shared="true" percentValue="false" includeChildSelections="true" includeChildForces="false"/>
+                    <condition type="instanceOf" value="1" field="selections" scope="ancestor" childId="23a4-cb25-4fc5-5c3c" shared="true"/>
```
Comment (nstephenh) [inline 2022 - Horus Heresy.gst:272]: Calling out the characters directly works.

File: 2022 - Legiones Astartes.cat [hunk 4]
```diff
@@ -42812,7 +42812,7 @@ It is not know if this is a mistake or not, as previously like with the Mech boo
               <modifiers>
                 <modifier type="increment" field="d2ee-04cb-5f8a-2642" value="25">
                   <repeats>
-                    <repeat field="selections" scope="23e4-c64b-65b4-c8a5" value="1" percentValue="false" shared="true" includeChildSelections="true" includeChildForces="false" childId="model" repeats="1" roundUp="false"/>
+                    <repeat field="selections" scope="23e4-c64b-65b4-c8a5" value="1" percentValue="false" shared="true" includeChildSelections="true" includeChildForces="false" childId="4b49-fbfe-8078-cae8" repeats="1" roundUp="false"/>
```
Comment (nstephenh) [inline 2022 - Legiones Astartes.cat:42815]: So we were counting gunners before, that explains it.

File: 2022 - Legiones Astartes.cat [hunk 5]
```diff
@@ -46800,7 +46803,7 @@ Additionally, at the end of any Phase in which a model with Gorgon Terminator ar
             <conditionGroup type="and">
               <conditions>
                 <condition type="equalTo" value="0" field="selections" scope="force" childId="ae4a-f95c-968e-eb46" shared="true" includeChildSelections="true"/>
-                <condition type="equalTo" value="1" field="selections" scope="parent" childId="21c3-2f28-7820-e51a" shared="true" percentValue="false" includeChildSelections="true" includeChildForces="false"/>
+                <condition type="equalTo" value="1" field="selections" scope="force" childId="21c3-2f28-7820-e51a" shared="true" percentValue="false" includeChildSelections="true" includeChildForces="false"/>
```
Comment (nstephenh) [inline 2022 - Legiones Astartes.cat]: This is good, though also there are a few duplicates of this condition. I can clean them up.

**Review by nstephenh — DISMISSED** [review]
With the small fixes I added, this looks good to me.

**Review by nstephenh — APPROVED** [review]
Looks good to me, I made a few small tweaks on top.

**Pull request merged**